packages/e2e-tests/specs/editor/various/inserter.test.js

```diff
@@ -20,4 +24,27 @@ describe( 'Inserter', () => {
 		const isPreviewVisible = await preview.isIntersectingViewport();
 		expect( isPreviewVisible ).toBe( true );
 	} );
+
+	it( 'last-inserted block should be given and keep the focus', async () => {
+		await page.focus( '.block-editor-default-block-appender__content' );
+		await page.keyboard.type( 'Testing inserted block focus' );
+
+		await openGlobalBlockInserter();
+
+		await page.keyboard.down( 'Control' );
+		await page.click( '.editor-block-list-item-image' );
+		await page.keyboard.up( 'Control' );
+
+		await page.waitForSelector( 'figure[data-type="core/image"]' );
+
+		const imageBlock = ( await getAllBlocks() ).find(
+			( block ) => block.name === 'core/image'
+		);
+
+		const selectedBlock = await page.evaluate( () => {
+			return wp.data.select( 'core/block-editor' ).getSelectedBlock();
+		} );
+
+		expect( selectedBlock.clientId ).toBe( imageBlock.clientId );
```

Comment:
Since we're only inserting one image block, this could be simplified to

```suggestion
		const selectedBlock = await page.evaluate( () => {
			return wp.data.select( 'core/block-editor' ).getSelectedBlock();
		} );

		expect( selectedBlock.name ).toBe( 'core/image' );
```

(Inspired by code in https://github.com/WordPress/gutenberg/blob/0596f1ba178e6e01f987fc89e9f098f2896206d9/packages/e2e-tests/specs/editor/various/writing-flow.test.js)